google/stackdriver_monitoring/metric_collector.py

```diff
@@ -17,23 +17,81 @@
 
 # pylint: disable=missing-docstring
 import argparse
+import collections
 import logging
 import logging.config
-import collections
+import socket
+import threading
+import time
 
 from http_server import (HttpServer, StdoutRequestHandler)
 
 import metric_collector_handlers as handlers
 import stackdriver_handlers as stackdriver_handlers
 
-from spectator_client import SpectatorClient
-from stackdriver_client import StackdriverClient
+import spectator_client
+import stackdriver_client
 
 
 HandlerDefinition = collections.namedtuple(
   'HandlersDefinition', ['handler', 'url_path', 'command_name', 'description'])
 
 
+class DummyMetricService(object):
+  def store(self, service_metrics):
+    spectator_client.foreach_metric_in_service_map(service_metrics, self.dump)
+
+  def dump(self, service, name, metadata, instance):
+    name, tags = spectator_client.normalize_name_and_tags(
+        name, metadata, instance)
+    if tags is None:
+      print 'IGNORE {0}.{1}'.format(service, name)
```

Comment:
no, this is not stackdriver related at all.
I dont know what you mean by format.
Here I'm just ignoring metrics that have no tags. This is a spectator thing. It is picking up a number of spring metrics which are mostly not interesting and those that are have other equivalents. The spring metrics cause problems because they have names that stackdriver doesnt like.

I put this in just as a placeholder because I dont have a stackdriver implementation here. I threw it out a while ago because it became obsolete. The current stackdriver implementation is in kork-stackdriver that is java embedded into the runtime services.

I'm preparing this server to make it accomodating to other metric services and perhaps another stab at stackdriver here with a different strategy for dealing with custom metrics.